### PR TITLE
🌱 Add missing PROD_REGISTRY variable to makefile

### PR DESCRIPTION
**Pull request by MarcelMue**
**What this PR does / why we need it**:

This variable was previously referenced but not set.

### CODE REVIEW

File: exp/operator/Makefile [hunk 1]
```diff
@@ -51,6 +51,7 @@ KUSTOMIZE := $(abspath $(TOOLS_BIN_DIR)/kustomize)
 
 # Define Docker related variables. Releases should modify and double check these vars.
 REGISTRY ?= gcr.io/$(shell gcloud config get-value project)
+PROD_REGISTRY ?= k8s.gcr.io/cluster-api-operator
```
[the remark this answers is not in the document]
Comment (ashish-amarnath) [reply exp/operator/Makefile:54]: Is there a naming convention we should be following?
Considering 
```
STAGING_REGISTRY ?= gcr.io/k8s-staging-cluster-api
```
Shouldn't this be
```
PROD_REGISTRY ?= gcr.io/k8s-prod-cluster-api
```

[the remark this answers is not in the document]
Comment (sbueringer) [reply exp/operator/Makefile:54]: Please ignore my comment above (https://github.com/kubernetes-sigs/cluster-api/pull/4409#discussion_r604371671)

Should we really use another registry for this images or re-use the regular one we have for the core and kubeadm controllers?
They would be:
```bash
STAGING_REGISTRY ?= gcr.io/k8s-staging-cluster-api
PROD_REGISTRY ?= k8s.gcr.io/cluster-api
```

fyi other providers:

CAPO
```bash
STAGING_REGISTRY ?= gcr.io/k8s-staging-cluster-api-aws
PROD_REGISTRY := us.gcr.io/k8s-artifacts-prod/cluster-api-aws
```

CAPO (I think I hit a length limit somewhere with `-openstack`)
```bash
STAGING_REGISTRY := gcr.io/k8s-staging-capi-openstack
PROD_REGISTRY := us.gcr.io/k8s-artifacts-prod/capi-openstack
```

Given that we use the same staging registry, I think it's reasonable to use the same prod registry as the other controllers in the same repo.

[the remark this answers is not in the document]
Comment (fabriziopandini) [reply exp/operator/Makefile:54]: Let's use the same registry we are using for CAPI (with a different image name)

**Pull request merged**